cmd/manager/main.go

```diff
@@ -225,3 +241,78 @@ func checkIfRequiredFilesExist(requiredFiles []string) error {
 	}
 	return nil
 }
+
+// clusterConfigChecks checks for required configuration in a cluster before starting the controller.
+// If the cluster doesn`t meet the configuration, it returns an error.
+func clusterConfigChecks(cfg *rest.Config) error {
+	// Get openshift api config client.
+	clienset, err := configclient.NewForConfig(cfg)
+	if err != nil {
+		return fmt.Errorf("error creating clientset %v", err)
+	}
+
+	// Check for valid K8s version. Currently any version >= 1.18 is supported.
+	// Check for Openshift version is not required.
+	if err := isValidKubernetesVersion(clienset); err != nil {
+		return fmt.Errorf("error in k8s version of the cluster %v", err)
+	}
+	// Check for determining if the cluster uses ovn-kubernetes
+	if err = isValidNetworkType(clienset); err != nil {
+		return fmt.Errorf("error in network type of the cluster %v", err)
+	}
+	//Check for determining if hybrid overlay is enabled
+	if err = isHybridOverlayEnabled(cfg); err != nil {
+		return fmt.Errorf("error in hybrid overlay %v", err)
+	}
+
+	return nil
+}
+
+// isValidKubernetesVersion checks for valid k8s version in the cluster. It returns an err for all versions below
+// v1.18.0
+func isValidKubernetesVersion(clientset *configclient.Clientset) error {
+	versionInfo, err := clientset.Discovery().ServerVersion()
+	if err != nil {
+		return err
+	}
+
+	k8sVersion := versionInfo.GitVersion
+	if strings.Compare(k8sVersion, baseK8sVersion) < 0 {
+		return fmt.Errorf("invalid server version: %v.Supported versions >= %v", k8sVersion,
+			baseK8sVersion)
+	}
+	return nil
+}
+
+//isValidNetworkType checks if cluster is using ovn-kubernetes
+func isValidNetworkType(clientset *configclient.Clientset) error {
```

Comment:
Is it possible to set this up so that we are checking for a set of network types that we support? And then have a set of network specific checks. So a factory pattern with a struct that has a validate function might be better.